Cannot view plugin info for UI plugins

**Describe the bug**

In the "Plugins > List Installed Plugins" page, clicking the "i" Info button for a UI plugin does not work.  The loading spinner appears and never resolves.  A 500 error is logged in the JS console.

**My Rundeck detail**

* Rundeck version: 3.3.0 SNAPSHOT, 3.2.x

**To Reproduce**
Steps to reproduce the behavior:
1. Go to 'System Config menu > Plugins >Installed Plugins'
2. Click on 'View All' type selector, choose "UI"
3. Click on the "Info" button ("i" icon) for a UI plugin
4. See loading spinner that does not resolve.

**Expected behavior**
Plugin information is displayed


**Additional context**

Stacktrace from server:

```
Caused by: java.lang.NullPointerException: Cannot get property 'value' on null object
	at rundeck.controllers.PluginController.pluginDetail(PluginController.groovy:189) ~[main/:?]
	... 79 more
```

--- a/rundeckapp/grails-app/controllers/rundeck/controllers/PluginController.groovy
+++ b/rundeckapp/grails-app/controllers/rundeck/controllers/PluginController.groovy
@@ -2,6 +2,7 @@ package rundeck.controllers
 
 import com.dtolabs.rundeck.app.support.PluginResourceReq
 import com.dtolabs.rundeck.core.authorization.AuthContext
+import com.dtolabs.rundeck.plugins.rundeck.UIPlugin
 import org.rundeck.core.auth.AuthConstants
 import com.dtolabs.rundeck.core.plugins.PluginValidator
 import com.dtolabs.rundeck.core.plugins.configuration.PluginAdapterUtility
@@ -185,7 +186,7 @@ class PluginController extends ControllerBase {
         def desc = null
         def instance = null
         if(service== "UI") {
-            desc = pluginService.listPlugins(UIPluginProviderService,uiPluginService.uiPluginProviderService).find { it.key == pluginName }.value.description
+            desc = pluginService.getPluginDescriptor(pluginName, uiPluginService.uiPluginProviderService)?.description
         } else {
             def pDescriptor = pluginService.getPluginDescriptor(pluginName, service)
             instance = pDescriptor?.instance

--- a/rundeckapp/grails-app/controllers/rundeck/controllers/PluginController.groovy
+++ b/rundeckapp/grails-app/controllers/rundeck/controllers/PluginController.groovy
@@ -2,7 +2,6 @@ package rundeck.controllers
 
 import com.dtolabs.rundeck.app.support.PluginResourceReq
 import com.dtolabs.rundeck.core.authorization.AuthContext
-import com.dtolabs.rundeck.plugins.rundeck.UIPlugin
 import org.rundeck.core.auth.AuthConstants
 import com.dtolabs.rundeck.core.plugins.PluginValidator
 import com.dtolabs.rundeck.core.plugins.configuration.PluginAdapterUtility

--- a/rundeckapp/src/test/groovy/rundeck/controllers/PluginControllerSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/controllers/PluginControllerSpec.groovy
@@ -1,7 +1,13 @@
 package rundeck.controllers
 
 import com.dtolabs.rundeck.core.common.Framework
-
+import com.dtolabs.rundeck.core.common.IFramework
+import com.dtolabs.rundeck.core.plugins.PluginMetadata
+import com.dtolabs.rundeck.plugins.rundeck.UIPlugin
+import com.dtolabs.rundeck.plugins.util.DescriptionBuilder
+import com.dtolabs.rundeck.server.plugins.services.UIPluginProviderService
+import grails.util.Described
+import org.apache.logging.log4j.core.config.plugins.util.PluginManager
 import org.grails.plugins.testing.GrailsMockMultipartFile
 import com.dtolabs.rundeck.core.plugins.ServiceProviderLoader
 import com.dtolabs.rundeck.core.plugins.ValidatedPlugin
@@ -282,7 +288,7 @@ class PluginControllerSpec extends Specification implements ControllerUnitTest<P
         setup:
         controller.frameworkService = Mock(FrameworkService)
         messageSource.addMessage("request.error.unauthorized.title",Locale.ENGLISH,"Unauthorized")
-        
+
         when:
         def pluginUrl = Thread.currentThread().getContextClassLoader().getResource(PLUGIN_FILE)
         params.pluginUrl = pluginUrl.toString()
@@ -292,5 +298,40 @@ class PluginControllerSpec extends Specification implements ControllerUnitTest<P
         1 * controller.frameworkService.getAuthContextForSubject(_)
         1 * controller.frameworkService.authorizeApplicationResourceType(_,_,_) >> false
         response.text == '{"err":"Unauthorized"}'
+    }
+
+    def "plugin detail for ui plugin"() {
+        given:
+
+            params.name = 'test1'
+            params.service = 'UI'
+            controller.pluginService = Mock(PluginService)
+            controller.uiPluginService = Mock(UiPluginService)
+            controller.pluginApiService = Mock(PluginApiService)
+            def uiPluginProviderService = Mock(UIPluginProviderService)
+            controller.frameworkService = Mock(FrameworkService)
+
+            def plugin = Mock(UIPlugin)
+            def description = DescriptionBuilder.builder().name('test1').build()
+            def pluginMeta = Mock(PluginMetadata)
+        when:
+            controller.pluginDetail()
+        then:
+            response.status == 200
+
+            response.contentType.startsWith 'application/json'
+            
+            1 * controller.uiPluginService.getUiPluginProviderService() >> uiPluginProviderService
+            1 * controller.uiPluginService.getPluginMessage('UI', 'test1', 'plugin.title', _, _) >> 'ptitle'
+            1 * controller.uiPluginService.getPluginMessage('UI', 'test1', 'plugin.description', _, _) >>
+            'pdescription'
+            1 * controller.pluginApiService.pluginPropertiesAsMap('UI', 'test1', _) >> []
+            1 * controller.pluginService.getPluginDescriptor('test1', uiPluginProviderService) >>
+            new DescribedPlugin<UIPlugin>(plugin, description, 'test1')
+            1 * controller.frameworkService.getRundeckFramework() >> Mock(IFramework) {
+                1 * getPluginManager() >> Mock(ServiceProviderLoader) {
+                    1 * getPluginMetadata('UI', 'test1') >> pluginMeta
+                }
+            }
     }
 }

